tests/resources/Docker-Util.robot

```diff
@@ -51,6 +53,44 @@ Get Container IP
     Should Be Equal As Integers  ${rc}  0
     [Return]  ${ip}
 
+# Use host docker daemon to start a dind container alongside this test container
+Start Docker Daemon On TCP
+    [Arguments]  ${docker-params}  ${dind-version}=docker:1.13-dind
+    ${rc}  ${id}=  Run And Return Rc And Output  docker run -it --privileged ${docker-params} -d ${dind-version}
+    Should Be Equal As Integers  ${rc}  0
+    [Return]  ${id}
+
+# The local dind version is embedded in Dockerfile
+# docker:1.13-dind
+# If you are running this keyword in a container, make sure it is run with --privileged turned on
+Start Docker Daemon Locally
+    [Arguments]  ${dockerd-params}  ${dockerd-path}=/usr/local/bin/dockerd-entrypoint.sh  ${log}=/var/log/daemon-local.log
+    File Should Exist  ${dockerd-path}
+    ${handle}=  Start Process  ${dockerd-path} ${dockerd-params} >${log} 2>&1  shell=True
+    Process Should Be Running  ${handle}
+    :FOR  ${IDX}  IN RANGE  5
+    \   ${pid}=  Run  ps aux | grep '[d]ockerd ' | awk '{print $2}' # Space after [d]ockerd in grep is essential to filter shell process
+    \   Run Keyword If  '${pid}' != '${EMPTY}'  Set Test Variable  ${dockerd-pid}  ${pid}
+    \   Exit For Loop If  '${pid}' != '${EMPTY}'
+    \   Sleep  1s
+    [Return]  ${handle}  ${dockerd-pid}
+    
+Cleanup Container On ID
+    [Arguments]  ${id}
+    ${rc}  ${output}=  Run And Return Rc And Output  docker stop ${id}
+    Should Be Equal As Integers  ${rc}  0
+    Should Be Equal  ${output}  ${id}
+    ${rc}  ${output}=  Run And Return Rc And Output  docker rm ${id}
+    Should Be Equal As Integers  ${rc}  0
+    Should Be Equal  ${output}  ${id}
+
+Kill Local Docker Daemon
+    [Arguments]  ${handle}  ${dockerd-pid}
+    Terminate Process  ${handle}
+    Process Should Be Stopped  ${handle}
+    ${rc}=  Run And Return Rc  kill -9 ${dockerd-pid}
+    Should Be Equal As Integers  ${rc}  0
+
```

Comment:
because it's started by dockerd binary not 'service docker start'. 'service docker start' doesn't seem to work properly. besides, host docker socket is mapped into container, we can use it directly by -H unix:///var/run/docker.sock. This is what "Start Docker Daemon Over Host Socket" does.
added dockerd binary is trying to give us the ability to start a daemon with customized parameter(like --insecure-registry) running locally(inside drone test container) on unix:///var/run/docker-local.sock